package/MDAnalysis/analysis/rdf.py

```diff
@@ -250,30 +342,30 @@ def _prepare(self):
         count, edges = np.histogram([-1], **self.rdf_settings)
         count = count.astype(np.float64)
         count *= 0.0
-        self.count = count
-        self.edges = edges
-        self.bins = 0.5 * (edges[:-1] + edges[1:])
+        self.results.count = count
+        self.results.edges = edges
+        self.results.bins = 0.5 * (edges[:-1] + edges[1:])
 
         # Need to know average volume
         self.volume = 0.0
         # Set the max range to filter the search radius
         self._maxrange = self.rdf_settings['range'][1]
 
-
     def _single_frame(self):
         pairs, dist = distances.capped_distance(self.g1.positions,
                                                 self.g2.positions,
                                                 self._maxrange,
                                                 box=self.u.dimensions)
         # Maybe exclude same molecule distances
         if self._exclusion_block is not None:
-            idxA, idxB = pairs[:, 0]//self._exclusion_block[0], pairs[:, 1]//self._exclusion_block[1]
+            idxA = pairs[:, 0]//self._exclusion_block[0],
```

Comment:
is the `,` necessary here?

Reply:
I think it's actually incorrect because now `idxA` is a tuple. If this still passes tests then I'd like to know why.